### ISSUE TITLE
format("WW") returns 52 instead of 02

### ISSUE THREAD

**Issue by arikhativa** [moment/moment#5969]
**Describe the bug**
Formatting a date with format("WW") returns an unexpected result.

**To Reproduce**
1. create this moment object -> var date = moment("2022-01-02")
2. use the format function -> date.format("WW")
3. function returned "52" (but this date "2022-01-02" is on the 2nd week of the year)

**Expected behavior**
function should return "02"

**Desktop (please complete the following information):**
 - OS: ubuntu 18
 - Version 2.10.2

**Moment-specific environment**
using nodejs 

Run these cmds on on the same setting as the bug reproduced"
console.log((new Date()).toString())
    Sun Jan 23 2022 10:59:24 GMT+0000 (UTC)
console.log((new Date()).toLocaleString())
    2022-1-23 10:59:24
console.log((new Date()).getTimezoneOffset())
    0
console.log(moment.version)
    2.10.2

**Comment by arikhativa** [comment]
it might be an issue with these functions:
https://github.com/moment/moment/blob/e96809208c9d1b1bbe22d605e76985770024de42/moment.js#L1346-L1374

**Comment by arikhativa** [comment]
btw date.week() is working just fine (returns "02")

**Comment by DiscoNova** [comment]
There are two things that need addressing here;

First, according to ISO-8601 (the international standard for dates) defines that:

- first day of a week is monday (and last day of week is sunday) and...
- first week of a year is the one that has first thursday of January in it

This somewhat unintuitively for countries that traditionally consider a week to start on sunday instead means that, 02-JAN-2022 actually falls on the last week of 2021 ... i.e. "week 52". Therefore, the result for `moment.format('WW')` is indeed correct in this case.

Secondly, the `moment.week()`-method is locale-dependent. The `EN_US`-locale (and a few others) defines that:

- first day of week is sunday (and last day of week is saturday) and...
- first week of a year is the one that has January 1st in it

This means that ... indeed, in this locale, 02-JAN-2022 falls on second week of 2022 ... i.e. "week 02". Therefore, the result for `moment.week()` is _also_ indeed correct in this case.

In conclusion, there seems to be no bug and the package works exactly as intended. Depending on what You're after, You should get (using Your currently applied locale) the following results:

- `moment.format('WW')` => `"52"`
- `moment.isoWeek()` => `52 /* Number */`
- `moment.week()` => `2 /* Number */`
- `moment.format('w')` => `"2"`
- `moment.format('ww')` => `"02"`

Hope this helps.

**Comment by arikhativa** [comment]
@DiscoNova Thanks!

just another small question, I actually used the format() function this way -> format("YYYY-WW")
and it returned -> 2022-52
so maybe "WW" is not buged, but there might be an issue here

code:
`
var date = moment("2022-01-02");
console.log(date.format("YYYY-WW"))
`

**Comment by DiscoNova** [comment]
I personally always go with the ISO-8601 dates, so now I'm just pulling rabbits out of my hat, but..

    let date = moment("2022-01-02"); console.log(date.format("GGGG-ww")); /* => "2021-52"? */

Would something like that work? According to the documentation of the format tokens, it might, but... I frankly don't know (and didn't have time to check)

**Comment by arikhativa** [comment]
@DiscoNova  ah i see

I tested and got:
console.log(now.format("GGGG-WW")) -> 2021-52
console.log(now.format("gggg-ww")) -> 2022-02

Thanks a lot :)
I'll close the ticket
